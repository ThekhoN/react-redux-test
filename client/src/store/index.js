import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from '../reducer';
import ReduxThunk from 'redux-thunk';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
// const store = createStore(rootReducer, composeEnhancers());
const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

export default store;
