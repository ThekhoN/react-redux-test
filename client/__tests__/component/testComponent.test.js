import React from 'react';
import TestComponent from '../../src/component/TestComponent';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import store from '../../src/store';
import {Provider} from 'react-redux';


test('TestComponent is a div element with class .test-component', () => {
  const component = renderer.create(<Provider store={store}><TestComponent /></Provider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  describe('TestComponent', () => {
    it('TestComponent is a div element with class .test-component', () => {
      const test = shallow(<Provider store={store}><TestComponent /></Provider>);
      expect(test.find('div.test-component')).to.have.length(1);
    });
  });
});
