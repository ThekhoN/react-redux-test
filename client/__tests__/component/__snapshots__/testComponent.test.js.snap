// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent is a div element with class .test-component 1`] = `
<div
  className="test-component"
>
  <span
    className="test-text"
  >
    This is a Connected Test Component
  </span>
  <h2>
    test: 
    <span
      className="special"
    >
      true
    </span>
  </h2>
  <br />
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
